core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -445,6 +443,11 @@ private void updateCacheStream(long blockId) throws IOException {
       return;
     }
 
+    // Don't cache the block if this block is already read from a remote worker.
```

Comment:
Hmm but I think we want to cache if there is a local worker, otherwise data will never be in more than 1 Alluxio worker?